src/Aspire.Hosting/ApplicationModel/IResourceBuilder.cs

```diff
@@ -25,14 +25,16 @@ public interface IResourceBuilder<out T> where T : IResource
     /// <typeparam name="TAnnotation">The type of the annotation to add.</typeparam>
     /// <param name="behavior">The behavior to use when adding the annotation.</param>
     /// <returns>The resource builder instance.</returns>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("ApiDesign", "RS0026:Do not add multiple public overloads with optional parameters", Justification = "<Pending>")]
     IResourceBuilder<T> WithAnnotation<TAnnotation>(ResourceAnnotationMutationBehavior behavior = ResourceAnnotationMutationBehavior.Append) where TAnnotation : IResourceAnnotation, new() => WithAnnotation(new TAnnotation(), behavior);
```

Comment:
I think that there is an argument that we could get rid of this overload. It isn't used that much because folks almost always want to pass in the annotation object.